Description: Fix FTBFS on armel when using armv4/armv4t instruction set.
Author: Evgeni Golov <evgeni@golov.de>
Debian-Bug: https://bugs.debian.org/598931

Index: unstable/egg/egg-fixed.c
===================================================================
--- unstable.orig/egg/egg-fixed.c	2015-05-25 16:51:17.410277336 -0400
+++ unstable/egg/egg-fixed.c	2015-05-25 16:55:29.122599888 -0400
@@ -705,7 +705,7 @@
      */
     int t = 0;
     int sh = 0;
-#ifndef __arm__
+#if !defined(__arm__) || defined(__ARM_ARCH_4T__) || defined(__ARM_ARCH_4__)
     unsigned int mask = 0x40000000;
 #endif
     unsigned fract = x & 0x0000ffff;
@@ -720,7 +720,7 @@
 	/*
 	 * Find the highest bit set
 	 */
-#if __arm__
+#if defined(__arm__) && !defined(__ARM_ARCH_4T__) && !defined(__ARM_ARCH_4__)
 	/* This actually requires at least arm v5, but gcc does not seem
 	 * to set the architecture defines correctly, and it is I think
 	 * very unlikely that anyone will want to use egg on anything
